src/macaron/policy_engine/prelude/helper_rules.dl

```diff
@@ -12,6 +12,38 @@ check_passed(component_id, check_name) :- check_result(_, check_name, 1, compone
 .decl check_failed(component_id: number, check_name: symbol)
 check_failed(component_id, check_name) :- check_result(_, check_name, 0, component_id).
 
+
+/**
+ * This relation provides the passed checks together with the confidence score for a component.
+   Parameters:
+   component_id: number
+       The target software component id.
+   check_name: symbol
+       The check name that matches the ``^mcn_([a-z]+_)+([0-9]+)$`` regular expression.
+   confidence: float
+       The confidence score computed for the check result.
+ */
+.decl check_passed_with_confidence(component_id: number, check_name: symbol, confidence: float)
+check_passed_with_confidence(component_id, check_name, confidence) :-
+    check_result(check_id, check_name, 1, component_id),
+    check_facts(check_id, confidence, component_id,_,_).
```

Comment:
I am not sure I understand the usage of `check_id` here. The first column of Check Result and Check Fact seems to be the unique ids for two separate tables:
- https://github.com/oracle/macaron/blob/5c2dbec369528e47fb2272a3f97852287ba75397/src/macaron/database/table_definitions.py#L421
- https://github.com/oracle/macaron/blob/5c2dbec369528e47fb2272a3f97852287ba75397/src/macaron/database/table_definitions.py#L389

Reply:
Good catch, I had used the wrong column. I renamed the column name to `check_result_id` to avoid mistakes. I also found a type mismatch in the SQLAlchemy mapping that SQLAlchemy/mypy had failed to catch.

See commit [2ff81c5](https://github.com/oracle/macaron/pull/816/commits/2ff81c530dc8336b9247a91e22e0e7d8bac970af)